File: slt_telecome-frontend/lib/models/employee.dart
class Employee {
  final int? empid;
  final String fullname;
  final String number;
  final String department;
  final String username;
  final String password;

  Employee({
    this.empid,
    required this.fullname,
    required this.number,
    required this.department,
    required this.username,
    required this.password,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      empid: json['empid'],
      fullname: json['fullname'],
      number: json['number'],
      department: json['department'],
      username: json['username'],
      password: json['password'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fullname': fullname,
      'number': number,
      'department': department,
      'username': username,
      'password': password,
    };
  }
}

File: slt_telecom/models/Employee.php
<?php

class Employee {
    public $empid;
    public $fullname;
    public $number;
    public $department;
    public $username;
    public $password;

    public function __construct($fullname, $number, $department, $username, $password, $empid = null) {
        $this->empid = $empid;
        $this->fullname = $fullname;
        $this->number = $number;
        $this->department = $department;
        $this->username = $username;
        $this->password = password_hash($password, PASSWORD_DEFAULT); // Hash the password for security
    }
}
?>
